utils/version.py

```diff
@@ -41,8 +42,10 @@ def get_version(obj=None):
     If obj is None, the version will be retrieved from the current appliance
 
     """
-    if isinstance(obj, Version) or isinstance(obj, float):
+    if isinstance(obj, Version):
         return obj
+    if not isinstance(obj, string_types):
+        obj = str(obj)
```

Comment:
We have decided to just convert obj to string in case it is not already a string, This way it can be then inspected with startswith as a normal string and then passed to Version.